src/Microsoft.AspNetCore.JsonPatch/JsonPatchDocumentOfT.cs

```diff
@@ -705,61 +705,57 @@ IList<Operation> IJsonPatchDocument.GetOperations()
             return allOps;
         }
 
-        private string GetPath<TProp>(Expression<Func<TModel, TProp>> expr)
+        // Internal for testing
+        internal string GetPath<TProp>(Expression<Func<TModel, TProp>> expr, string position)
         {
-            return "/" + GetPath(expr.Body, true).ToLowerInvariant();
+            var segments = GetPath(expr.Body, true);
+            var path = String.Join("/", segments);
+            if (position != null)
+            {
+                path += "/" + position;
+                if (segments.Count == 0)
+                {
+                    return path.ToLowerInvariant();
+                }
+            }
+
+            return "/" + path.ToLowerInvariant();
         }
 
-        private string GetPath(Expression expr, bool firstTime)
+        private List<string> GetPath(Expression expr, bool firstTime)
```

Comment:
It wouldn't hurt to rename this to something like `GetPathSegments` its usage isn't really interchangable with `GetPath` that returns a string